SmartcarAuth/SmartcarAuth.swift

```diff
@@ -72,8 +72,8 @@ public class SmartcarAuth: NSObject {
     public func generateLink(for oem: OEMName) -> String {
         var stateString = ""
         
-        if self.request.state.characters.count > 0 {
-            stateString = "&state=" + self.request.state
+        if let state = self.request.state {
+            stateString = "&state=" + state
```

Comment:
I agree with the comment on url encoding/decoding this param. 
Other changes lgtm